framework/src/controller/controller.js

```diff
@@ -228,12 +232,17 @@ class Controller {
 
 		const child = child_process.fork(program, parameters);
 
-		child.on('exit', (code, signal) => {
+		this.childrenList.push(child);
+
+		child.on('exit', async (code, signal) => {
 			this.logger.error(
 				`Module ${moduleAlias}(${name}:${version}) exited with code: ${code} and signal: ${signal}`
 			);
-			// TODO: Reload child instead of exiting the parent process
-			process.exit(1);
+
+			await this.bus.unregisterChannel(moduleAlias);
+
+			// Reload the module
+			this._loadChildProcessModule(alias, Klass, options);
```

Comment:
Once child process is `exit`, we must not load it again at least for this release. We can develop more dedicated reload strategy later.

Reply:
This was discussed and decided to reload the faulty module. It can be improved in next versions but it works pretty well as it is.

Reply:
No its not the right solution, it can lead into infinite reloading and exhaust the system memory.

Reply:
For current release its good if parent process exit wit proper logs if child process is crashed. Later we can develop a strategy, how to reload modules, or even have configurable options with module load retry etc. This topic needed to have more thought process and a refined solution.

Reply:
We can improve the solution later on, but the resilience should be there.